modules/tinymce/src/themes/silver/test/ts/browser/editor/SilverPopupSinkBoundsTest.ts

```diff
@@ -110,26 +112,231 @@ describe('browser.tinymce.themes.silver.editor.SilverPopupSinkBoundsTest', () =>
-        ui_mode: 'split'
-      },
-      () => setupElement(false),
-      []
-    );
+    const marginOfError = 5;
 
-    it('TINY-9226: popup needs to be inside scrolling container', async () => {
-      const editor = hook.editor();
-      assert.isFalse(editor.isHidden());
+    context('DOM', () => {
+      const hook = TinyHooks.bddSetupFromElement<Editor>(
+        {
+          ...sharedSettings,
+          ui_mode: 'split'
+        },
+        () => setupElement(false),
+        []
+      );
 
-      const marginOfError = 5;
-      const menuRect = await pOpenAndMeasureMenu(editor);
-      const wrapperRect = measureWrapper(editor);
+      it('TINY-9226: popup needs to be inside scrolling container', async () => {
+        const editor = hook.editor();
+        assert.isFalse(editor.isHidden());
+
+        const menuRect = await pOpenAndMeasureMenu(editor);
+        const wrapperRect = measureWrapper(editor);
+
+        // Compare the menu to the scrolling container
+        assert.isAbove(menuRect.top, wrapperRect.top - marginOfError, 'Menu should not appear above scrolling container');
+        assert.isBelow(menuRect.bottom, wrapperRect.bottom + marginOfError, 'Menu should not below scrolling container');
+      });
+    });
+
+    interface Scenario {
+      readonly label: string;
+      readonly getRect: (rect: DOMRect) => {
+        readonly clientX: number;
+        readonly clientY: number;
+      };
+    }
+
+    context('ShadowDOM', () => {
+      const nestedLevel = 15;
+
+      const setupRoot = (shadowParent: SugarElement<HTMLElement>): SugarElement<HTMLElement | ShadowRoot> => {
+        const shadowHost = SugarElement.fromTag('div', document);
+        Class.add(shadowHost, 'test-shadow-root');
+        Insert.append(shadowParent, shadowHost);
+        return SugarElement.fromDom(shadowHost.dom.attachShadow({ mode: 'open' }));
+      };
+
+      const setupSingleScrollerElement = (shadow: string) => {
+        const scroller = SugarElement.fromTag('div');
+        Css.setAll(scroller, {
+          overflow: 'auto',
+          height: '500px',
+          margin: '0px 200px', // 200px to make sure the bounds are not calculated from the viewport
+          outline: '3px solid brown'
+        });
+        Class.add(scroller, 'scrolling-wrapper');
+
+        const target = SugarElement.fromTag('textarea');
+
+        if (shadow === 'outside') {
+          // The scroller is being outside of the ShadowRoot
+          const root = setupRoot(scroller);
+          Insert.append(
+            root,
+            target
+          );
+          Insert.append(SugarBody.body(), scroller);
+        } else {
+          // The scroller is being appended to the ShadowRoot
+          const root = setupRoot(SugarBody.body());
+          Insert.append(scroller, target);
+          Insert.append(root, scroller);
+        }
+
+        return {
+          element: target,
+          teardown: () => {
+            Remove.remove(scroller);
```

Comment:
When the scroller is outside the shadow root, we have to remove the root on teardown I believe